src/libraries/System.Net.Security/src/System/Net/Security/SslConnectionInfo.Unix.tt

```diff
@@ -0,0 +1,124 @@
+            DataKeySize = (int)s_cipherStrengthEnumValues[(data >> (16 - (4 * 3)) & 0xF)];
+            DataHashAlg = (int)s_hashEnumValues[(data >> (16 - (4 * 4)) & 0xF)];
+            DataHashKeySize = GetHashSize((HashAlgorithmType)DataHashAlg);
+
+            static int GetHashSize(HashAlgorithmType hash)
+            {
+                switch (hash)
+                {
+                    case HashAlgorithmType.None:
+                        return 0;
+                    case HashAlgorithmType.Md5:
+                        return 128;
+                    case HashAlgorithmType.Sha1:
+                        return 160;
+                    case HashAlgorithmType.Sha256:
+                        return 256;
+                    case HashAlgorithmType.Sha384:
+                        return 384;
+                    case HashAlgorithmType.Sha512:
+                        return 512;
+                    default:
+                        throw new ArgumentOutOfRangeException(nameof(hash));
+                }
+            }
+        }
+
+        private static ReadOnlySpan<int> s_exchangeAlgorithmTypes =>
+            new[] { <#
+foreach (ExchangeAlgorithmTypeIndex val in exchangeEnumValues)
+{
+#>(int)ExchangeAlgorithmType.<#= val #>, <#
+}
+#>};
+
+        private static ReadOnlySpan<int> s_cipherEnumValues =>
+            new[] { <#
+foreach (CipherAlgorithmTypeIndex val in cipherEnumValues)
+{
+#>(int)CipherAlgorithmType.<#= val #>, <#
+}
+#>};
+
+        private static ReadOnlySpan<int> s_cipherStrengthEnumValues =>
+            new[] { <#
+foreach (CipherAlgorithmStrengthIndex val in cipherStrengthEnumValues)
+{
+#>(int)CipherAlgorithmStrength.<#= val #>, <#
+}
+#>};
+
+        private static ReadOnlySpan<int> s_hashEnumValues =>
+            new[] { <#
+foreach (HashAlgorithmTypeIndex val in hashEnumValues)
+{
+#>(int)HashAlgorithmType.<#= val #>, <#
+}
+#>};
+
+        private static ushort GetPackedData(TlsCipherSuite cipherSuite)
+        {
+            switch (cipherSuite)
+            {
+<#
+foreach (TlsCipherSuite val in tlsEnumValues)
+{
+TlsCipherSuiteData data = new CipherSuiteNameData(val.ToString()).Data;
+byte exchangeAlgorithmType =  (byte)Enum.Parse<ExchangeAlgorithmTypeIndex>(EnumHelpers.ToFrameworkName(data.KeyExchangeAlgorithm));
+byte cipherAlgorithmType =  (byte)Enum.Parse<CipherAlgorithmTypeIndex>(EnumHelpers.ToFrameworkName(data.CipherAlgorithm));
+byte cipherAlgorithmStrength = (byte)Enum.Parse<CipherAlgorithmStrengthIndex>(EnumHelpers.CipherAlgorithmStrengthIndexToName(data.CipherAlgorithmStrength));
+byte hashAlgorithmType =  (byte)Enum.Parse<HashAlgorithmTypeIndex>(EnumHelpers.ToFrameworkName(data.MACAlgorithm));
+#>
+                case TlsCipherSuite.<#= val #>: return <#= exchangeAlgorithmType #> << (16 - (4 * 1)) | <#= cipherAlgorithmType #> << (16 - (4 * 2)) | <#= cipherAlgorithmStrength #> << (16 - (4 * 3)) | <#= hashAlgorithmType #> << (16 - (4 * 4));
```

Comment:
Nit: I think having the shift amounts as direct numbers like `(... << 12) | (... << 8) | (... << 4) | ....` would look just fine. I do not think that the expressions help with readability.